compiler/scalc/ast.ml

```diff
@@ -56,37 +56,29 @@ and naked_expr =
     }
   | EArray of expr list
   | ELit of lit
+  | EPosLit
   | EApp of { f : expr; args : expr list }
-  | EAppOp of { op : operator Mark.pos; args : expr list }
+  | EAppOp of { op : operator Mark.pos; args : expr list; tys : typ list }
   | EExternal of { modname : VarName.t Mark.pos; name : string Mark.pos }
 
 type stmt =
   | SInnerFuncDef of { name : VarName.t Mark.pos; func : func }
   | SLocalDecl of { name : VarName.t Mark.pos; typ : typ }
   | SLocalInit of { name : VarName.t Mark.pos; typ : typ; expr : expr }
-  | SLocalDef of { name : VarName.t Mark.pos; expr : expr; typ : typ }
-  | STryWEmpty of { try_block : block; with_block : block }
-  | SRaiseEmpty
-  | SFatalError of Runtime.error
+  | SLocalDef of { name : VarName.t Mark.pos; typ : typ; expr : expr }
+  | SFatalError of { pos_expr : expr; error : Runtime.error }
   | SIfThenElse of { if_expr : expr; then_block : block; else_block : block }
   | SSwitch of {
       switch_var : VarName.t;
       switch_var_typ : typ;
       enum_name : EnumName.t;
       switch_cases : switch_case list;
     }
-  | SReturn of naked_expr
-  | SAssert of naked_expr
+  | SReturn of expr
+  | SAssert of { pos_expr : expr; expr : expr }
```

Comment:
Should put a comment here saying that `pos_expr` is a the reifed position as an expression.